spec/tasks/init_spec.rb

```diff
@@ -143,4 +143,64 @@
       expect(stub).to have_been_requested.once
     end
   end
+
+  context 'when an empty response body is received' do
+    before(:each) do
+      stub_request(:post, 'http://0.0.0.0/post')
+        .to_return(status: response_status, body: nil)
+    end
+
+    let(:opts) {
+      {
+        method:   'post',
+        base_url: "#{url}/post"
+      }
+    }
+
+    [101, 204, 304].each do |iter_status|
+      describe "and the response status is #{iter_status}" do
+        let(:response_status) { iter_status }
+        it 'gracefully returns nil' do
+          result = handler.task(opts)
+          expect(result.keys).to match_array(success_keys)
+          expect(result[:body]).to be_nil
+        end
+      end
+    end
+
+    # According to rfc2616, an empty-body 200 reponse like this is forbidden,
+    # so this test condition should never happen.  I added these 'specs' for
+    # comparison with the behavior from 204 + friends.
+    #
+    # I don't know how the task should handle the HTTP server responding with
+    # an empty body for response that MUST NOT have one according to rfc22616;
+    # It's out of spec, so I'm included to let it fail.
+    #
+    # However:
+    #
+    # * Under these (forbidden) conditions, the task only fails when
+    #   json_endpoint = true.
+    # * Should the json_endpoint logic handle this weird case more gracefully?
+    # * Should default (non-json_endpoint) logic be forced to fail?
+    #
```

Comment:
Since this task doesn't enforce whether or not certain response types have a body, I think we can remove this test. I'm not particularly keen on enforcing properly formatted responses, either, since that's likely something that should happen server-side.

We could potentially just ignore response bodies when there shouldn't be one, though rfc2616 does not seem to offer guidance on this like it does when receiving a request that must not have a body (unless I missed that in my scan). I lean towards that being separate work, just to get this improvement in quicker and give us an opportunity to discuss what the right approach is.

Reply:
Sounds great; out it all goes!